src/dodal/beamlines/i22.py

```diff
@@ -38,6 +39,7 @@
 )
 
 
+@device_factory(lazy=True)
 def saxs(
     wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
 ) -> PilatusDetector:
```

Comment:
Expected something like:
```python
@device_factory(lazy=True)
def saxs() -> PilatusDetector:
    return PilatusDetector(
        f"{BL}-EA-PILAT-01:",
        drv_suffix="CAM:",
        hdf_suffix="HDF5:",
        directory_provider=get_directory_provider(),
    )
```

Reply:
ok thanks for the use case, now the code can be tested against it